src/Compilers/CSharp/Portable/Symbols/Retargeting/RetargetingNamedTypeSymbol.cs

```diff
@@ -393,17 +402,104 @@ public sealed override bool AreLocalsZeroed
         internal sealed override bool IsRecordStruct => _underlyingType.IsRecordStruct;
         internal sealed override bool HasPossibleWellKnownCloneMethod() => _underlyingType.HasPossibleWellKnownCloneMethod();
 #nullable enable
-        internal sealed override bool IsExtension => false; // PROTOTYPE
+        internal sealed override bool IsExtension => _underlyingType.IsExtension;
+        internal sealed override bool IsExplicitExtension => _underlyingType.IsExplicitExtension;
+
+        internal sealed override TypeSymbol? ExtendedTypeNoUseSiteDiagnostics
+        {
+            get
+            {
+                if (ReferenceEquals(_lazyExtendedType, ErrorTypeSymbol.UnknownResultType))
+                {
+                    TypeSymbol? acyclicExtendedType = GetDeclaredExtensionUnderlyingType();
```

Comment:
>GetDeclaredExtensionUnderlyingType( [](http://example.com/codeflow?start=54&length=35)

Since ```GetDeclaredExtensionUnderlyingType``` substitutes an extension type with an error type, do we still need the cycle detection logic below?

Reply:
This can be simplified indeed. Thanks